api/tests/core/mails/transactional/bookings/beneficiary_booking_cancellation_test.py

```diff
@@ -0,0 +1,241 @@
+                offer__id=123456,
+            ),
+        )
+
+        # When
+        email_data = get_beneficiary_booking_cancellation_email_data(booking.individualBooking)
+
+        # Then
+        assert email_data == {
+            "Mj-TemplateID": 1091464,
+            "Mj-TemplateLanguage": True,
+            "Vars": {
+                "can_book_again": 1,
+                "event_date": "26 novembre 2019",
+                "event_hour": "19h29",
+                "is_event": 1,
+                "is_free_offer": 0,
+                "offer_id": "AHREA",
+                "offer_name": "Test event name",
+                "offer_price": "10.20",
+                "user_first_name": "Fabien",
+            },
+        }
+
+    @override_features(ENABLE_SENDINBLUE_TRANSACTIONAL_EMAILS=False)
+    def test_should_return_is_free_offer_when_offer_price_equals_to_zero(self):
+        # Given
+        booking = booking_factories.CancelledIndividualBookingFactory(stock__price=0)
+
+        # When
+        email_data = get_beneficiary_booking_cancellation_email_data(booking.individualBooking)
+
+        # Then
+        assert email_data["Vars"]["is_free_offer"] == 1
+
+    @override_features(ENABLE_SENDINBLUE_TRANSACTIONAL_EMAILS=False)
+    def test_should_return_the_price_multiplied_by_quantity_when_it_is_a_duo_offer(self):
+        # Given
+        booking = booking_factories.CancelledIndividualBookingFactory(quantity=2, stock__price=10)
+
+        # When
+        email_data = get_beneficiary_booking_cancellation_email_data(booking.individualBooking)
+
+        # Then
+        assert email_data["Vars"]["offer_price"] == "20.00"
+
+
+@pytest.mark.usefixtures("db_session")
+class SendSendiblueBeneficiaryBookingCancellationEmailTest:
+    @override_features(ENABLE_SENDINBLUE_TRANSACTIONAL_EMAILS=True)
+    def test_should_called_send_email_with_valid_data(self):
+        # given
+        booking = booking_factories.IndividualBookingFactory()
+
+        # when
+        send_beneficiary_booking_cancellation_email(booking.individualBooking)
+
+        # then
+        assert mails_testing.outbox[0].sent_data["template"] == {
+            "id_prod": 223,
+            "id_not_prod": 33,
+            "tags": ["jeunes_offre_annulee_jeune"],
+        }
+        assert mails_testing.outbox[0].sent_data["To"] == booking.individualBooking.user.email
+        params_key_list = [
+            "CAN_BOOK_AGAIN",
+            "EVENT_DATE",
+            "EVENT_HOUR",
+            "IS_EVENT",
+            "IS_FREE_OFFER",
+            "OFFER_NAME",
+            "OFFER_PRICE",
+            "USER_FIRST_NAME",
+            "OFFER_LINK",
+        ]
+        for params_key in params_key_list:
+            assert params_key in mails_testing.outbox[0].sent_data["params"].keys()
+
+
+@pytest.mark.usefixtures("db_session")
+class MakeBeneficiaryBookingCancellationEmailSendinblueDataTest:
+    @override_features(ENABLE_SENDINBLUE_TRANSACTIONAL_EMAILS=True)
```

Comment:
Same ici, `@override_features(ENABLE_SENDINBLUE_TRANSACTIONAL_EMAILS=True)` peut décorer la classe plutôt

Reply:
E   TypeError: override_features cannot be used to decorate a class